export `create` method

jonschlinkert opened the pull request:
just getting this going 

cc @cowboy / @tkellen

jonschlinkert committed:
link to docs

jonschlinkert committed:
adds `create` method as top level export

jonschlinkert committed:
remove unnecessary dep

jonschlinkert committed:
pass grunt, log, verbose on options

jonschlinkert committed:
remove code examples

jonschlinkert committed:
make variable consistent with other legacy libs

cowboy commented on index.js:
```diff
@@ -7,256 +7,210 @@
+var _ = require('lodash');
+
+exports.create = function(options) {
+  options = _.extend({}, {grunt: null, log: null}, options);
```
Just expect a `grunt` option, `log.wordlist`, `log.error` `verbose` and `template.process` will be pulled from it.

cowboy replied:
After merging options, maybe we can explicitly test that the passed-in `grunt` has the API this module needs, and throw if not. Eg.: 

``` js
if (!options.grunt.log) { throw new Error('missing options.grunt.log'); }
if (!options.grunt.log.wordlist) { throw new Error('missing options.grunt.log.wordlist'); }
...
```

cowboy replied:
Strike the exception throwing, that won't work. It _could_ work if we asserted in each method, though.

cowboy commented on index.js:
```diff
@@ -7,256 +7,210 @@
+   */
+
+  config.init = function(obj) {
+    options.verbose.write('Initializing config...').ok();
```
Change all `options.verbose` to `options.grunt.verbose`.

cowboy commented on index.js:
```diff
@@ -7,256 +7,210 @@
+    var p = legacyUtil.pluralize;
+    var props = _.toArray(arguments).map(config.getPropString);
+    var msg = 'Verifying propert' + p(props.length, 'y/ies') +
+      ' ' + options.log.wordlist(props) + ' exist' + p(props.length, 's') +
```
Change all `options.log` to `options.grunt.log`.

cowboy commented on test/test.js:
```diff
@@ -10,12 +10,18 @@
-    this.origData = config.data;
+    config = legacyConfig.create({
+      grunt: grunt,
+      log: grunt.log,
```
Remove this and the following line.

jonschlinkert committed:
removes grunt as a dependency:

- replace log utils methods with `grunt-legacy-log-utils`
- use `grunt-legacy-template` for process method
- use facade from `grunt-legacy-event-logger` to setup logging/event methods

jonschlinkert committed:
updates docs, adds changelog

jonschlinkert committed:
use grunt-legacy-log-facade

jonschlinkert committed:
fix deps
